core/src/test/scala/unit/kafka/server/CreateTopicsRequestWithPolicyTest.scala

```diff
@@ -65,14 +70,23 @@ class CreateTopicsRequestWithPolicyTest extends AbstractCreateTopicsRequestTest
       Map("policy-topic1" -> error(Errors.POLICY_VIOLATION, Some("Topics should have at least 5 partitions, received 4"))))
 
     validateErrorCreateTopicsRequests(new CreateTopicsRequest.Builder(
-      Map("policy-topic2" -> new CreateTopicsRequest.TopicDetails(4, 1.toShort)).asJava, timeout, true).build(),
+      Map("policy-topic2" -> new CreateTopicsRequest.TopicDetails(4, 3.toShort)).asJava, timeout, true).build(),
       Map("policy-topic2" -> error(Errors.POLICY_VIOLATION, Some("Topics should have at least 5 partitions, received 4"))))
 
+    val configs = Map(LogConfig.RetentionMsProp -> 5001.toString)
+    validateErrorCreateTopicsRequests(new CreateTopicsRequest.Builder(
+      Map("policy-topic3" -> new CreateTopicsRequest.TopicDetails(11, 2.toShort, configs.asJava)).asJava, timeout, true).build(),
+      Map("policy-topic3" -> error(Errors.POLICY_VIOLATION, Some("RetentionMs should be less than 5000ms if replicationFactor > 5"))))
+
+    validateErrorCreateTopicsRequests(new CreateTopicsRequest.Builder(
+      Map("policy-topic4" -> new CreateTopicsRequest.TopicDetails(11, 3.toShort, Map.empty.asJava)).asJava, timeout, true).build(),
+      Map("policy-topic4" -> error(Errors.POLICY_VIOLATION, Some("RetentionMs should be less than 5000ms if replicationFactor > 5"))))
+
     val assignments = replicaAssignmentToJava(Map(0 -> List(1), 1 -> List(0)))
     validateErrorCreateTopicsRequests(new CreateTopicsRequest.Builder(
-      Map("policy-topic3" -> new CreateTopicsRequest.TopicDetails(assignments)).asJava, timeout).build(),
-      Map("policy-topic3" -> error(Errors.POLICY_VIOLATION,
-        Some("""Topic partitions should have at least 2 partitions, received 1 for partition 0"""))), checkErrorMessage = true)
+      Map("policy-topic5" -> new CreateTopicsRequest.TopicDetails(assignments)).asJava, timeout).build(),
+      Map("policy-topic5" -> error(Errors.POLICY_VIOLATION,
+        Some("Topic partitions should have at least 2 partitions, received 1 for partition 0"))), checkErrorMessage = true)
```

Comment:
Any reason to explicitly enable `checkErrorMessage` here and not the others? The default is `true`.